docs/topics/classes.md

```diff
@@ -1,245 +1,747 @@
+}
+```
+
+The visibility of the constructor will be public, meaning it can be accessed from anywhere.
 If you don't want your class to have a public constructor, declare an empty primary constructor with non-default visibility:
 
 ```kotlin
-class DontCreateMe private constructor() { /*...*/ }
+class Person private constructor() { /*...*/ }
 ```
 
-> On the JVM, if all of the primary constructor parameters have default values, the compiler will generate an additional parameterless constructor which will use the default values. This makes it easier to use Kotlin with libraries such as Jackson or JPA that create class instances through parameterless constructors.
+> On the JVM, if all primary constructor parameters have default values, the compiler generates 
+> an additional parameterless constructor that uses those default values. 
+> 
+> This makes it easier to use Kotlin with libraries such as [Jackson](https://github.com/FasterXML/jackson) 
+> or [Spring Data JPA](https://spring.io/projects/spring-data-jpa), which create class objects through 
+> parameterless constructors.
 >
+> In the following example, Kotlin automatically generates a parameterless constructor `Person()` that uses the default value `""`:
+> 
 > ```kotlin
-> class Customer(val customerName: String = "")
+> class Person(val personName: String = "")
 > ```
 >
 {style="note"}
 
-## Creating instances of classes
+## Creating objects of classes
 
-To create an instance of a class, call the constructor as if it were a regular function. You can assign the created instance to a [variable](basic-syntax.md#variables):
+Like in other object-oriented programming languages, a class in Kotlin serves as a blueprint or template 
+that defines the properties (attributes) and behaviors (functions) of objects. When you create an instance 
+of a class, you are creating a concrete object based on that blueprint.
+
+To create an object of a class, use the class name followed by parentheses (`()`), similar to calling a function:
 
 ```kotlin
-val invoice = Invoice()
+// Creates an object of the 'Person' class
+val person1 = Person()
+```
+
+In Kotlin, you can create objects:
 
-val customer = Customer("Joe Smith")
+* **Without arguments** (`()`): creates an object using a constructor with default values.
+* **With arguments** (`("value")`): creates an object by passing specific values to the constructor.
+
+As you can see, you can assign the created object to a mutable (`var`) or immutable (`val`) [variable](basic-syntax.md#variables):
+
+```kotlin
+// Creates an object by passing a specific value and assigns it to a mutable variable
+var person2 = Person("Joe")
+
+// Creates an object using the default constructor's value and assigns it to an immutable variable
+val person1 = Person()
 ```
 
-> Kotlin does not have a `new` keyword.
->
-{style="note"}
+Typically, you create objects within the `main()` function because it serves as the entry point of your Kotlin program. 
```

Comment:
This statement is not true, you create objects whenever you need to do that, and not only within the `main` function